src/chain/chain.rs

```diff
@@ -307,6 +308,17 @@ impl Chain {
         Ok(vec![self.state.new_info.clone()])
     }
 
+    /// Starts pruning as requested. Returns a new `SectionInfo`.
+    pub fn prune(&mut self) -> Result<SectionInfo, RoutingError> {
+        if self.state.change != PrefixChange::None {
+            log_or_panic!(LogLevel::Warn, "Tring to prune during pfx change.",);
+        }
+
+        self.state.change = PrefixChange::Pruning;
```

Comment:
What is the purpose of  PrefixChange::Pruning in the context of this PR? are we not setting it to immediatly reset it?

Reply:
Ah, this is a leftover from my previous approach.
I shall remove it in the context of `immediately reset`.